### DISCUSSION TITLE
how to use the channel of serial connection?

### DISCUSSION THREAD

**Discussion by murph12F** [bluenviron/gomavlib#133] [Q&A]
Hello, i am trying to write a little code that will connect via serial and will send some mavlink commands, in the examples i only managed to find how to write once we received something before, since it needs a channel and it seems that in all examples the channel is always took from waiting an event and then using the channel of that event to send back a message, i ll leave a link to an example file to better understand. 

https://github.com/bluenviron/gomavlib/blob/main/examples/endpoint-serial/main.go

I want to send an event even without receiving nothing from the serial port, so i guess i have to create a channel, but what i am not understanding is how to use the channel created by the serial connection if it actually creates one, and if not how to properly create a channel. i ll share my draft code. 

```go


.....
	node, err := gomavlib.NewNode(gomavlib.NodeConf{
		Endpoints: []gomavlib.EndpointConf{
			gomavlib.EndpointSerial{
				Device: "/dev/ttyUSB0",
				Baud:   57600,
			},
		},
		Dialect:     common.Dialect,
		OutVersion:  gomavlib.V2,
		OutSystemID: 10,
	})
	if err != nil {
		panic(err)
	}
	defer node.Close()

	mex := &common.MessagePositionTargetGlobalInt{
		TimeBootMs:      1234567,
		CoordinateFrame: 2,
		TypeMask:        0,
		LatInt:          512345678,
		LonInt:          -123456789,
		Alt:             150.75,
		Vx:              12.34,
		Vy:              -5.67,
		Vz:              1.23,
		Afx:             0.5,
		Afy:             -0.3,
		Afz:             0.1,
		Yaw:             45.0,
		YawRate:         -0.1,
	}

	//what to put instead of nil to use as channel? 
	node.WriteMessageTo(nil, mex)

}
```

thank you very much.

**Comment by aler9** [comment]
Hello, use `WriteMessageAll` instead of `WriteMessageTo`.

In case of a serial endpoint, a channel is created automatically. You can obtain it by listening on `node.Events()` as displayed in all examples. But since there's always one channel, it's useless to store it in the software. Just use `WriteMessageAll`.
